_posts/2021-06-09-#21800.md

```diff
@@ -0,0 +1,94 @@
+---
+layout: pr
+date: 2021-06-09
+title: "Mempool/validation: mempool ancestor/descendant limits for packages"
+pr: 21800
+authors: [glozow]
+components: ["mempool", "tests", "validation"]
+host: glozow
+status: upcoming
+commit:
+---
+
+## Notes
+
+* Apart from coinbase transactions, every Bitcoin transaction has at least one _parent_, another
+  transaction whose outputs are spent as its inputs. We call the spending transaction the _child_.
+_Ancestors_ include parents, their parents, and so forth. _Descendants_ include children, their
+children, and so forth.
+
+* Each mempool policy has information on its ancestors and descendants in the mempool. This speeds
+  up some operations (i.e. selecting which transactions to include in a block and which to evict for
+low feerates) and slows down others (adding and removing transactions).
+
+* We have ancestor/descendant limits as part of our mempool policy with the goal of constraining the
+  maximum computation needed for such operations. For example, if seeing a conflicting transaction
+in a block could cause us to remove a transaction with 10,000 descendants in the mempool, our
+mempool could end up a sink of computational resources (which could also be triggered intentionally
+by an attacker) rather than a useful cache of unconfirmed transactions.
+
+* The current ancestor/descendant limits are: no transaction can have more than 25 (or user config)
+  or 101KvB of ancestors + descendants. The mempool member function
```

Comment:
"can have more than 25 or 101KvB of ancestors + descendants" is slightly confusing. It's valid to have 15 ancestors and 15 descendants, which the sentence implies is invalid.